### DISCUSSION TITLE
useHydrateAtoms does not report hydration errors in local dev

### DISCUSSION THREAD

**Discussion by bahtiyara** [pmndrs/jotai#2750] [Bug report]
### Bug Description

I used to use Provider without any store, but in this way I cannot be able to use atom values outside the component. So I added the store to my Provider, but by doing that, React reported some hydration errors, like this 👇

![image](https://github.com/user-attachments/assets/af95897e-1633-4b8d-b887-faea74e97cb8)

I understand why hydration error occurs, but I don't understand why it has something to do with using store or not. Or I could assume is hydration errors have been suppressed by default in store-less mode?

Another big problem is the hydration errors only show up in the remote server, and not in the local dev environment, which makes it extremely difficult to debug.

### Reproduction Link

example.com

**Comment by dai-shi** [comment]
@akkadaska is this something you are familiar with?

**Comment by bahtiyara** [comment]
At the end of the day, I fixed the problem myself by replacing the useHydrateAtom hook with my own hook:

```jsx
type Item<AtomValue = any> = [PrimitiveAtom<AtomValue>, AtomValue]

export function useOwnHydrateAtom(...items: Item[]) {
	const has = useRef(false)

	if (!has.current) {
		for (const [atom, value] of items) {
			rootStore.set(atom, value)
		}

		has.current = true
	}
}
```

The code above removed the errors for me.

**Comment by akkadaska** [comment]
@bahtiyara 
I could help you. Could you show me some reproduction project?
And could you tell me your understanding about why hydration error occurs?
